Generated unique-service-path should not be changed after compilation

ayeshLK opened the pull request:
## Purpose
> $subject

Fixes [#1813](https://github.com/ballerina-platform/ballerina-standard-library/issues/1813)

## Examples

## Checklist
- [X] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

shafreenAnfar commented on compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/service/path/ServicePathGenerator.java:
```diff
@@ -0,0 +1,33 @@
+ * {@code ServicePathGenerator} generates unique service path for subscriber service and saves into a resource file.
+ */
+public interface ServicePathGenerator {
+    void generate(PackageId packageId, Path currentProjectRoot, int serviceId) throws ServicePathGeneratorException;
```
Since this is a public interface let's document the methods as well.

ayeshLK replied:
Resolved with [0b8acfc51547940274ac74b3341d55e47b01fb2c](https://github.com/ballerina-platform/module-ballerina-websub/pull/359/commits/0b8acfc51547940274ac74b3341d55e47b01fb2c)

shafreenAnfar commented:
Why haven't we updated the change log ?

shafreenAnfar commented:
Does it work for `bal run <project>` as well ?

chamil321 commented on ballerina/sub_listener.bal:
```diff
@@ -51,8 +51,13 @@ public class Listener {
         self.listenerConfig = self.httpListener.getConfig();
         self.port = self.httpListener.getPort();
         self.gracefulShutdownPeriod = config.gracefulShutdownPeriod;
+        check self.externInit();
```
```suggestion
        return self.externInit();
```

chamil321 commented on compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/service/path/AbstractServicePathGenerator.java:
```diff
@@ -0,0 +1,65 @@
+public abstract class AbstractServicePathGenerator implements ServicePathGenerator {
+    @Override
+    public void generate(PackageId packageId, Path currentProjectRoot, int serviceId)
+            throws ServicePathGeneratorException {
```
```suggestion
        throws ServicePathGeneratorException {
```

chamil321 commented on native/src/test/resources/testng.xml:
```diff
@@ -11,17 +11,17 @@
+            <class name="io.ballerina.stdlib.websub.ServiceInfoRetrieverTest"/>
+        </classes>
+    </test>
+</suite>
```
```suggestion
</suite>

```

ayeshLK commented:
> Why haven't we updated the change log ?

Fixed with [3e8bda66f38eac28562d84e71046a9f95ab60897](https://github.com/ballerina-platform/module-ballerina-websub/pull/359/commits/3e8bda66f38eac28562d84e71046a9f95ab60897)

> Does it work for `bal run <project>` as well ?

When we use `bal run <project>` it uses the generated thin-jars. But currently ballerina project API does not have a life-cycle event which triggers after thin-jar creation (Issue [#32717](https://github.com/ballerina-platform/ballerina-lang/issues/32717)). Hence currently this does not work. In the proposal I have included a future work section [1] which describes an alternative approach which we could use after compile time code-injection support is introduced.

[1] - [https://github.com/ballerina-platform/module-ballerina-websub/blob/master/docs/proposals/service-path-generation.md#future-work](https://github.com/ballerina-platform/module-ballerina-websub/blob/master/docs/proposals/service-path-generation.md#future-work)
